cdap-security/src/main/java/co/cask/cdap/security/authorization/DefaultAuthorizationEnforcementService.java

```diff
@@ -111,14 +113,45 @@ protected void runOneIteration() throws Exception {
 
   @Override
   public void enforce(EntityId entity, Principal principal, Action action) throws Exception {
+    enforce(entity, principal, Collections.singleton(action));
+  }
+
+  @Override
+  public void enforce(EntityId entity, Principal principal, Set<Action> actions) throws Exception {
     if (!authorizationEnabled) {
       return;
     }
-    Set<Privilege> privileges = cacheEnabled ? authPolicyCache.get(principal) : getPrivileges(principal);
-    privileges = privileges == null ? ImmutableSet.<Privilege>of() : privileges;
-    if (!privileges.contains(new Privilege(entity, action))) {
-      throw new UnauthorizedException(principal, action, entity);
+    Set<Privilege> privileges = getPrivileges(principal);
+    // If a principal has ALL privileges on the entity, authorization should succeed
+    if (privileges.contains(new Privilege(entity, Action.ALL))) {
+      return;
+    }
+    // Otherwise check for the specific action requested
+    for (Action action : actions) {
```

Comment:
Discussed in person. If the operation is a 'get' operation (let's say to view an entity on the UI), then we will check if *any* action on the entity is allowed. If the operation is a 'read', then we will enforce the READ action.